src/live_restore/live_restore_fs.c

```diff
@@ -1581,6 +1592,58 @@ __validate_live_restore_path(WT_FILE_SYSTEM *fs, WT_SESSION_IMPL *session, const
     return (fh->close(fh, (WT_SESSION *)session));
 }
 
+/*
+ * __wt_live_restore_fs_log_copy --
+ *     Copy the log files from the source to the destination prior to recovery. We also need to copy
+ *     over prep log files as logging will expect these are available to be used. This needs to
+ *     happen after __wt_logmgr_config to ensure the relevant logging configuration has been parsed.
+ */
+int
+__wt_live_restore_fs_log_copy(WT_SESSION_IMPL *session)
+{
+    WT_CONNECTION_IMPL *conn = S2C(session);
+    WT_DECL_RET;
+    if (!F_ISSET(&conn->log_mgr, WT_LOG_CONFIG_ENABLED))
+        return (0);
+
+    u_int logcount = 0;
+    char **logfiles = NULL;
+    WT_DECL_ITEM(filename);
+    WT_FH *fh = NULL;
+    uint32_t lognum;
+
+    /* Get a list of actual log files. */
+    WT_ERR(__wt_log_get_files(session, WT_LOG_FILENAME, &logfiles, &logcount));
+    for (u_int i = 0; i < logcount; i++) {
+        WT_ERR(__wt_log_extract_lognum(session, logfiles[i], &lognum));
+        /* Call log open file to generate the full path to the log file. */
+        WT_ERR(__wt_log_openfile(session, lognum, 0, &fh));
+        ret = __wti_live_restore_fs_fill_holes(fh->handle, (WT_SESSION *)session);
+        WT_TRET(__wt_close(session, &fh));
+        WT_ERR(ret);
+    }
+    WT_ERR(__wt_fs_directory_list_free(session, &logfiles, logcount));
+
+    /* Get a list of prep log files. */
```

Comment:
I think this can go away.

Reply:
The comment or the whole block of code?

Reply:
I meant the whole block of code. But with your explanation, please add a comment.